### DISCUSSION TITLE
How to generate server resolvers based on a gql schema?

### DISCUSSION THREAD

**Discussion by alucardu** [dotansimha/graphql-code-generator#8073] [Q&A]
I run a `apollo-server-express` server on my NodeJS env:

```typescript
// app.ts

import express from 'express';
import { ApolloServer } from 'apollo-server-express';
import resolvers from './resolvers.js';
import typeDefs from './schema.js';


const apolloServer = new ApolloServer({
  resolvers,
  typeDefs
});

async function startServer() {
    await apolloServer.start();
    const app = express();
    apolloServer.applyMiddleware({ app });
    app.listen(4000, function () {
      console.log(`🚀 Server rseady at http://localhost:4000${apolloServer.graphqlPath}`)
    });
}

startServer();
```
But I defined my `resolvers` and `typedefs` manually. 

```javascript
// resolvers.js

const resolvers = {
  Query: {
    getHello() {
      return 'hello'
    }
  },
}

module.exports = resolvers;
```
```javascript
// schema.js
const typeDefs = gql`
  type Query {
    getHello: String
  }
`;

module.exports = typeDefs;
```

I also defined a gql schema that I use to create React hooks:
```
// getHello.query.gql

query getHello {
  hello
}
```

So I was hoping I could also generate the server resolvers and typedefs using Codegen. I've tried using the `typescript-resolvers` plugin, but that only seems to generate types. Any suggestions?

**Comment by saihaj** [comment]
Only resolver types are generated. If you are looking to generate CRUD code that is not something codegen generates but there are other projects you should try out maybe https://redwoodjs.com which handles many things.
